### DISCUSSION TITLE
Can MLIR files be visualized with tools like netron?

### DISCUSSION THREAD

**Discussion by alan-chen-dongsheng** [iree-org/iree#18948] [Frontends] [answered]
Do we have any tools to visualize MLIR Files? LIke netron

MLIR seams had this feature by Graphviz https://github.com/llvm/llvm-project/blob/main/mlir/lib/Transforms/ViewOpGraph.cpp
But I'm not know how to using it! does somebody know how to using it?

**Comment by ScottTodd** [comment, marked as answer]
For general MLIR, you found `--view-op-graph`. Some references for that https://mlir.llvm.org/docs/Passes/#-view-op-graph (that's generated from the source file, so not much new information).

We had some bespoke tools at https://github.com/ScottTodd/iree-llvm-sandbox/tree/web-tools/web-tools

The "Model Explorer" project: https://github.com/google-ai-edge/model-explorer is a modern approach to visualizing model structures. That supports MLIR, but some engineering work would be needed to teach it how to use our dialects / program structures.

**Reply by alan-chen-dongsheng** [reply]
Thanks very much @ScottTodd ,That's is what i need.
